vignettes/Vaccines.Rmd

```diff
@@ -40,153 +123,262 @@ simparams <- get_parameters(list(
 
 ```{r}
-dose_data <- data.frame(timestep = output$timestep,
-                        dose = rep(c("mass 1", "mass 2", "mass 3", "mass booster"), each = length(output$timestep)),
-                        n = c(output$n_rtss_mass_dose_1, output$n_rtss_mass_dose_2, output$n_rtss_mass_dose_3,
-                              output$n_rtss_mass_booster_1))
-dose_data[dose_data$n > 0, ]
+# Use the get_parameters() function to generate a new parameter set with a seasonal profile with malaria incidence in children aged 0-5 rendered in the model output:
+seas_simparams <- get_parameters(
+	list(
+    	human_population = human_population,
+    	incidence_rendering_min_ages = 0,
+    	incidence_rendering_max_ages = 5 * year,
+    	individual_mosquitoes = FALSE,
+    	model_seasonality = TRUE, # Let's try a bi-modal model
+    	g0 = 0.285505,
+    	g = c(-0.325352,-0.0109352,0.0779865),
+    	h = c(-0.132815,0.104675,-0.013919)
+  )
+)
+
+seas_simparams <- set_equilibrium(seas_simparams, starting_EIR)
+
+# Run no vaccine scenario
+output_seas_control <- run_simulation(sim_length *2, seas_simparams)
+```
+
+Next, we will run the scenario with a seasonal mass vaccination campaign implemented starting 3.5 months before the peak of the transmission season.
+```{r}
+# Find the peak seasonality
+peak <- peak_season_offset(seas_simparams)
+
+seasmass_simparams <- set_mass_rtss(
+  parameters = seas_simparams,
+  timesteps = round(year + (peak - month * 3.5), 0),# The vaccination will begin 3.5 months prior to the peak seasonality in the second year.
+  coverages = 1, # 100% of the population between min_ages and max_ages is vaccinated.
+  min_ages = 5 * month, #
+  max_ages = 15 * year,
+  min_wait = 0, # There is no minimum wait between the last vaccination.
+  boosters = round(c(12 * month + 2 * month)), # The booster is given 14 months after the first dose. 
+  booster_coverage = 1) # 100% of the vaccinated population is boosted.
+
+output <- run_simulation(sim_length * 2, seasmass_simparams)
+```
+
+#### Plot clinical incidence
 
+```{r, fig.width = 6, fig.height = 4 fig.align = 'center'}
+plot_inci(type = 'seasonal')
 ```
 
+#### Plot prevalence
+```{r, fig.width = 6, fig.height = 4 fig.align = 'center'}
+plot_prev(type = 'seasonal')
+```
+
+#### Plot doses
+
+```{r, fig.width = 6, fig.height = 4, fig.align = 'center', message=FALSE}
+plot_doses()
+```
+
+
 ## RTS,S EPI
 
-You can opt for a more gradual dosing using the EPI strategy. 
-Individuals will be vaccinated once they reach 5 months. For this intervention
-we see a much more gradual increase in impact following implementation
+We can also opt to vaccinate using the EPI strategy, where individuals will be vaccinated once they reach a certain age. In the example below, individuals will be vaccinated once they reach 5 months of age. For this intervention, we see a much more gradual impact following implementation.
```

Comment:
Below, also show the default output of the doses variable to show that individuals begin the series at 5 months, but are really vaccinated at 5, 6.5, and 8 months for the primary series. Clarify that the model put the vaccine into effect once the third dose has been given. The booster is given 12 months following the THIRD dose